components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m

```diff
@@ -481,6 +508,91 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
   [self.view setNeedsLayout];
 }
 
+- (void)animateToPreferredContentHeight:(CGFloat)preferredContentHeight
+                           withDuration:(NSTimeInterval)duration
+                             completion:(void (^_Nullable)(BOOL))completion {
+  CGSize newPreferredContentSize =
+      CGSizeMake(CGRectGetWidth(self.view.bounds), preferredContentHeight);
+  if ([self isMobileLandscape]) {
+    self.contentViewController.preferredContentSize = newPreferredContentSize;
+    return;
+  }
+  CGPoint contentOffset =
+      [self calculateContentOffsetWithPreferredContentHeight:preferredContentHeight];
+  CGFloat precentageOfFullScreen =
+      [self precentageOfFullScreenWithPreferredContentHeight:preferredContentHeight];
+  BOOL shouldSetContentHeight = NO;
+  if (_contentVCPreferredContentSizeHeightCached > preferredContentHeight) {
+    shouldSetContentHeight = YES;
+  } else {
+    if (_contentVCPreferredContentSizeHeightCached != preferredContentHeight) {
```

Comment:
if they are equal, is anything needed to be done, or can we exit early like with landscape? also if nothing is needed then do we still need the contentOffset calculation method above?

Reply:
This will override drawer factor and animate to size